include/core/shape.h

```diff
@@ -755,6 +831,148 @@ class CompositeShape : public Shape<dim>
     operations;
 };
 
+template <int dim>
+class OpenCascadeShape : public Shape<dim>
+{
+public:
+  /**
+   * @brief Constructor for an OpenCascade shape
+   * @param file_name The name of the file describing the shape
+   * @param position The shape center
+   * @param orientation The shape orientation
+   */
+  OpenCascadeShape<dim>(std::string         file_name,
+                        const Point<dim> &  position,
+                        const Tensor<1, 3> &orientation)
+    : Shape<dim>(0.1, position, orientation)
+  {
+    // First, we read the shape file name and check if an effective radius as
+    // been given in addition to the file name
+    std::vector<std::string> shape_arguments_str_list(
+      Utilities::split_string_list(file_name, ";"));
+    local_file_name = shape_arguments_str_list[0];
+#ifdef DEAL_II_WITH_OPENCASCADE
+    // set the new effective radius if it was given with the file name.
+    if (shape_arguments_str_list.size() > 1)
+      {
+        this->effective_radius =
+          Utilities::string_to_double(shape_arguments_str_list[1]);
+      }
+
+    // Checks the file name extension to identify which type of OpenCascade
+    // shape we are working with.
+    std::size_t found_step   = local_file_name.find(".step");
+    std::size_t found_step_2 = local_file_name.find(".stp");
+    std::size_t found_igs    = local_file_name.find(".iges");
+    std::size_t found_igs_2  = local_file_name.find(".igs");
+    std::size_t found_stl    = local_file_name.find(".stl");
```

Comment:
I think you could find the position of the . in the string and get the extension with substr and use a switch case for the reader